### DISCUSSION TITLE
[resolved] Odroczona płatność - Termin spłaty

### DISCUSSION THREAD

**Discussion by Bober95** [allegro/allegro-api#4333] [Suggestions - resolved]
Witam,
Mam pytanie jak uzyskać termin spłaty faktury z odroczoną płatnością przez API ?
W zwracanych danych odnalazłem tylko że jest to odroczona płatność lecz żadnej informacji jaka ma widnieć na niej data
Pozdrawiam

**Comment by Lukasz-Zurek** [comment]
Nie uzyskasz terminu spłaty faktury z odroczoną płatnością przez API, ponieważ nie zwracamy tych danych w żadnym z naszych zasobów. Zwracamy jedynie informacje, o których wspomniałeś w poście. Przypominam, że podpisana umowa faktoringowa gwarantuje Ci, że pieniądze za towar otrzymasz od razu po wysłaniu lub doręczeniu go do klienta. Po wcześniejszym dodaniu faktury VAT. Natomiast Klient ma możliwość wyboru terminu zapłaty we własnym zakresie.

**Comment by Bober95** [comment]
To w takim razie jak poprawnie obsłużyć taką płatność ? 
W umowie występuje taki zapis:

> Sprzedający zobowiązuje się do przesłania Kupującemu i PragmaGO Faktury, w treści której
wskazane będą:
> b) termin płatności odpowiadający wybranemu przez Kupującego odroczonemu
terminowi płatności za Fakturę, tj. 21, 45 lub 60 dni licząc od dnia Transakcji,

Więc do automatyzacji procesu wystawiania FV potrzebuję tej wartości

**Comment by Lukasz-Zurek** [comment]
Wszelkie pytania dotyczące usługi faktoringowej i jej obsługi proszę kierować przez [formularz kontaktowy](https://allegro.pl/pomoc/kontakt?kategoria=c486d322-b6ca-4cb4-9995-fc1ffd340902&subjectId=9aed45c1-1c74-43a6-8989-e01da7a8597d).

**Comment by dlis2010** [comment]
Witam,

Spotkaliśmy się z tym samym problemem co @Bober95. Pełna obsługa odroczonej płatności przez API jest niemożliwa - wystawianie faktur musi odbywać się w sposób ręczny.

Odroczona płatność od strony API to jeden typ płatności, a tak naprawdę powinny być trzy różne dla: 
- odroczona płatność - 21 dni
- odroczona płatność - 45 dni
- odroczona płatność - 60 dni

Bardzo proszę o rozważenie rozwiązania, aby EXTENDED_TERM zamienić odpowiednio na EXTENDED_TERM_21, EXTENDED_TERM_45 oraz EXTENDED_TERM_60. Ewentualnie dodać do struktury payment informację o terminie spłaty.

**Comment by Lukasz-Zurek** [comment]
@dlis2010 Dziękuję za sugestie. Przekazuję je do działu odpowiedzialnego za rozwój REST API.

**Comment by rasskabak** [comment]
Wiadomo czy coś rusza się w tej sprawie? Jest to funkcja konieczna do obsługi odroczonej płatności, a od ostatniego komentarza minął już prawie miesiąc.

**Comment by Lukasz-Zurek** [comment]
@rasskabak Nie. Na tę chwilę nie mamy żadnych konkretnych informacji.

**Comment by AureliuszBrussy** [comment]
Wdrożyliśmy już zmianę i zwracamy już informację o terminie spłaty. Więcej informacji w [newsie](https://github.com/allegro/allegro-api/issues/4446).
